### PR TITLE
Add heliosynchronous orbits

### PR DESCRIPTION
**Pull request by jorgepiloto**
With this pull request, a new orbit creation method regarding `SSO orbits` is added. It contains also some tests coming from Vallado 11-2. 

The `if...elif...else` implementation was the only idea that came to my mind when working with these kind of orbits since just two of the three following parameters `{a, ecc, inc}` are needed in order to define them.

### CODE REVIEW

File: src/poliastro/twobody/orbit.py [hunk 1]
```diff
@@ -681,6 +681,99 @@ def geostationary(
...
+            True anomaly.
+        epoch : ~astropy.time.Time, optional
+            Epoch, default to J2000.
+        plane : ~poliastro.frames.Planes
+            Fundamental plane of the frame.
+        """
+
+        # Constants for Sun-Synchronours Orbits (SSO)
+        n_sunsync = 1.991063853e-7 * u.one / u.s
+        R_earth = Earth.R
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: Let's not hardcode the Earth, and use the `self.attractor` instead

File: src/poliastro/twobody/orbit.py [hunk 2]
```diff
@@ -681,6 +681,99 @@ def geostationary(
...
+        nu : ~astropy.units.Quantity
+            True anomaly.
+        epoch : ~astropy.time.Time, optional
+            Epoch, default to J2000.
+        plane : ~poliastro.frames.Planes
+            Fundamental plane of the frame.
+        """
+
+        # Constants for Sun-Synchronours Orbits (SSO)
+        n_sunsync = 1.991063853e-7 * u.one / u.s
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: I guess this value is hardcoded for the Earth as well?

File: src/poliastro/twobody/orbit.py [hunk 3]
```diff
@@ -681,6 +681,99 @@ def geostationary(
...
+        R_earth = Earth.R
+        k_earth = Earth.k
+        J2_earth = Earth.J2
+
+        if (a is None) and (ecc is None) and (inc is None):
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: I think we can mostly reuse this logic as is

https://github.com/satellogic/orbit-predictor/blob/e05e847e5344f8099a6ddc444cb2c6dced919975/orbit_predictor/predictors/numerical.py#L146-L178

File: src/poliastro/twobody/orbit.py [hunk 4]
```diff
@@ -681,6 +681,99 @@ def geostationary(
...
+            Epoch, default to J2000.
+        plane : ~poliastro.frames.Planes
+            Fundamental plane of the frame.
+        """
+
+        # Constants for Sun-Synchronours Orbits (SSO)
+        n_sunsync = 1.991063853e-7 * u.one / u.s
+        R_earth = Earth.R
+        k_earth = Earth.k
+        J2_earth = Earth.J2
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: As I discussed with @Meuge privately a while ago:

> "Sun-synchronous orbits are possible around other oblate planets, such as Mars. But Venus, for example, is too spherical to have a satellite in sun-synchronous orbit. If 𝐽2 is too small, you can't get sufficient torque to rotate the orbit fast enough. That is the case for Venus, whose 𝐽2 is about 0.4% that of Earth's.  this lack of oblateness is due to Venus' extremely low rotation rate"

[Book](https://www.springer.com/gp/book/9783319034157)

Therefore, where possible I think we should consider raising an error?

File: src/poliastro/twobody/orbit.py [hunk 5]
```diff
@@ -681,6 +681,99 @@ def geostationary(
         altitude = geo_radius - attractor.R
         return cls.circular(attractor, altitude)
 
+    @classmethod
+    def heliosynchronous(
+        cls,
+        a=None,
+        ecc=None,
+        inc=None,
+        raan=0 * u.deg,
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: For orbit designers it's more interesting to set the LTAN/LTDN, Local Time on Ascending/Descending Node. There's some working, but incomplete, logic here:

https://github.com/satellogic/orbit-predictor/blob/e05e847e5344f8099a6ddc444cb2c6dced919975/orbit_predictor/predictors/numerical.py#L180-L183

**Comment by Juanlu001** [comment]
Also, this will need a rebase.

File: src/poliastro/twobody/orbit.py (hunk 1 above)
Comment (jorgepiloto) [reply src/poliastro/twobody/orbit.py]: Sure! :rocket:

**Comment by jorgepiloto** [comment]
Now that #746 has been merged I am taking this again :mag:

File: src/poliastro/twobody/orbit.py (hunk 2 above)
Comment (jorgepiloto) [reply src/poliastro/twobody/orbit.py]: Yes, directly taken from the `Handbook of Satellite Orbits`. However now that #746 has been merged, this is not necessary anymore.

**Comment by noc0lour** [comment]
Is this PR superseding #598 ?  I'd like to give this one a shot in some calculations.

**Comment by jorgepiloto** [comment]
Hi @noc0lour, in response to your question: 

> Is this PR superseding #598 ? I'd like to give this one a shot in some calculations.

first implementation to have a clean and approved code will merge. I was waiting till #763 was ready before updating this but feel completely free to modify this pull request or open yours if you want :smile:

**Comment by noc0lour** [comment]
I'm already on testing this with some existing codebase, trying to replace a custom simple/limited propagator with poliastro (:

**Comment by Juanlu001** [comment]
I think we can close this in favor of #774?

**Comment by jorgepiloto** [comment]
> I think we can close this in favor of #774?

Sure, no problem :rocket:

**Comment by jorgepiloto** [comment]
:x: Closing in favor of #774 :x:

**Pull request closed without merging**